3-enrich/stream-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/kinesis/KinesisEnrichApp.scala

```diff
@@ -378,4 +378,12 @@ class KinesisEnrichConfig(config: Config) {
   val maxBackoff = backoffPolicy.getLong("maxBackoff")
 
   val useIpAddressAsPartitionKey = outStreams.hasPath("useIpAddressAsPartitionKey") && outStreams.getBoolean("useIpAddressAsPartitionKey")
+  val partitionKey:Option[String] = if (outStreams.hasPath("partitionKey") && outStreams.getString("partitionKey").isEmpty) {
+    Some(null)
```

Comment:
A user might want a null partition key for keyless messaging, so if the partitionKey property is specified, but the value is an empty string, the key will be null.